src/api/app/controllers/webui/project_controller.rb

```diff
@@ -709,8 +709,8 @@ def add_maintained_project
 
   def remove_maintained_project
     authorize @project, :update?
-    @project.maintained_projects.delete(MaintainedProject.find_by(project: @maintained_project))
-    if @project.save
+    maintained_project = MaintainedProject.find_by(project: @maintained_project)
+    if maintained_project && @project.maintained_projects.destroy(maintained_project)
```

Comment:
`destroy` returns the object that was destroyed (which acts as `true`) if it succeeded and `false` otherwise. 

> There's a series of callbacks associated with destroy. If the `before_destroy` callback throws `:abort` the action is cancelled and destroy returns `false`. 

http://edgeapi.rubyonrails.org/classes/ActiveRecord/Persistence.html#method-i-destroy